account/views.py

```diff
@@ -0,0 +1,135 @@
+            or 
+            Account.objects.filter(email = data['email'])
+            ):
+            return JsonResponse({'message':'DUPLICATED_USER'}, status=401)
+
+        if data['is_male'] == True:
+            sex_value = '남'
+        else:
+            sex_value = '여'
+
+        # 회원정보 생성
+        Account(
+    	    email               = data['email'],
+            name    		    = data['name'],
+        	phone_number        = data['phone_number'],
+            date_of_birth       = YYYY_MM_DD,
+            sex                 = sex_value,
+            password            = bcrypt.hashpw(data['password'].encode('utf-8'), bcrypt.gensalt()).decode('utf-8'),
+            is_agreed_emailing  = True,
+            is_agreed_texting   = False,
+        ).save()
+        return JsonResponse({'message':'SUCCESS'}, status=200)
+        
+class SignInView(View):
+    def post(self, request):           
+        data = json.loads(request.body)
+
+        # 이메일 미입력    
+        if not data.get('email'):
+            return JsonResponse({'message':'KEY_ERROR'}, status=400)
+
+        # 비밀번호 미입력
+        if not data.get('password'):
+            return JsonResponse({'message':'KEY_ERROR'}, status=400)       
+
+        # 없는 이메일
+        if not Account.objects.filter(email=data['email']):
+            return JsonResponse({'message':'INVALID_USER'}, status=400)
+       
+        # 비밀번호 틀림
+        if not bcrypt.checkpw(
+            data['password'].encode('utf-8'), 
+            Account.objects.get(email = data['email']).password.encode('utf-8')
+            ):
+            return JsonResponse({'message':'INVALID_USER'}, status=401)
+
+        # 입력한 이메일과 비밀번호 일치하면 성공 - 토큰발행
+        accessed_user   = Account.objects.get(email=data['email'])
+        access_token = jwt.encode({'account_id': accessed_user.id}, SECRET_KEY, ALGORITHM)
+        return JsonResponse({'Authorization': access_token.decode('utf-8')}, status=200)
+        
+class WishlistView(View):
+    @login_required
+    def post(self, request):
+        data = json.loads(request.body)
+        if not 'id' in data.keys(): 
+            return JsonResponse({'message':'KEY_ERROR'}, status=401)
+
+        if not Product.objects.filter(id=data['id']): 
+            return JsonResponse({'message':'INVALID_PRODUCT_NUMBER'}, status=400)
+
+        if Wishlist.objects.filter(product_id = data['id'], account_id=request.account_id): #위시리스트에 이미 있는 경우 삭제
+            Wishlist.objects.get(product_id=data['id'], account_id=request.account_id).delete()
+            return JsonResponse({'message':'DELETED'}, status=200)
+
+        Wishlist.objects.create(product_id=data['id'], account_id=request.account_id) # 위시리스트에 없는 경우 생성
+        return JsonResponse({'message':'ADDED'}, status=200)
+
+    @login_required
+    def get(self, request):
+        data = json.loads(request.body)
+        wishitems = Wishlist.objects.filter(account_id=request.account_id)
```

Comment:
아래와 같이 prefetch_related를 사용할 경우, wishlist와 many_to_many 관계인 product 상품들을 한번에 가져와서 캐싱하기 때문에 129~134줄에서 wishitem.product.main_image를 하더라도 DB에 쿼리를 하지 않습니다.

prefetch_related(), select_related() 메소드를 공부해보세요~~
https://docs.djangoproject.com/en/3.1/ref/models/querysets/
```suggestion
        wishitems = Wishlist.objects.filter(account_id=request.account_id).prefetch_related('product_set')
```
![스크린샷 2020-09-21 오후 9 13 57](https://user-images.githubusercontent.com/53214510/93765157-5fe9df80-fc4f-11ea-8246-63b5f0630c1a.png)